patches/chromium/feat_add_support_for_missing_dialog_features_to_shell_dialogs.patch

```diff
@@ -218,168 +218,103 @@ index 9780c80ffff7bdb715a9adcb656e5d33be974db6..cf6edf5c68a8e2f8ffda119b58c6283b
+ ServiceAvailability g_service_availability = ServiceAvailability::kNotStarted;
  
-@@ -171,10 +176,18 @@ void SelectFileDialogLinuxPortal::StartAvailabilityTestInBackground() {
-   if (GetAvailabilityTestCompletionFlag()->IsSet())
++uint32_t g_available_portal_version = 0;
++uint32_t g_required_portal_version = kXdgPortalRequiredVersion;
++
+ scoped_refptr<base::SequencedTaskRunner>& GetMainTaskRunner() {
+   static base::NoDestructor<scoped_refptr<base::SequencedTaskRunner>>
+       main_task_runner;
+@@ -89,9 +97,10 @@ void OnGetPropertyReply(dbus::Response* response) {
      return;
+   }
+ 
+-  g_service_availability = version >= kXdgPortalRequiredVersion
++  g_service_availability = version >= g_required_portal_version
+                                ? ServiceAvailability::kAvailable
+                                : ServiceAvailability::kNotAvailable;
++  g_available_portal_version = version;
+ }
+ 
+ void OnServiceStarted(std::optional<bool> service_started) {
+@@ -159,6 +168,12 @@ void SelectFileDialogLinuxPortal::StartAvailabilityTestInBackground() {
+   }
+   g_service_availability = ServiceAvailability::kInProgress;
  
 +  auto* cmd = base::CommandLine::ForCurrentProcess();
-+  unsigned int xdg_portal_required_version;
 +  if (!base::StringToUint(cmd->GetSwitchValueASCII(kXdgPortalRequiredVersionFlag),
-+                          &xdg_portal_required_version)) {
-+    xdg_portal_required_version = kXdgPortalRequiredVersion;
++                          &g_required_portal_version)) {
++    g_required_portal_version = kXdgPortalRequiredVersion;
 +  }
 +
-   dbus_thread_linux::GetTaskRunner()->PostTask(
-       FROM_HERE,
-       base::BindOnce(
--          &SelectFileDialogLinuxPortal::CheckPortalAvailabilityOnBusThread));
-+          &SelectFileDialogLinuxPortal::CheckPortalAvailabilityOnBusThread,
-+          xdg_portal_required_version));
- }
+   GetMainTaskRunner() = base::SequencedTaskRunner::GetCurrentDefault();
  
- // static
-@@ -185,6 +198,11 @@ bool SelectFileDialogLinuxPortal::IsPortalAvailable() {
-   return is_portal_available_;
+   dbus_utils::CheckForServiceAndStart(dbus_thread_linux::GetSharedSessionBus(),
+@@ -175,6 +190,11 @@ bool SelectFileDialogLinuxPortal::IsPortalAvailable() {
+   return g_service_availability == ServiceAvailability::kAvailable;
  }
  
 +// static
-+int SelectFileDialogLinuxPortal::GetPortalVersion() {
-+  return available_portal_version_;
++uint32_t SelectFileDialogLinuxPortal::GetPortalVersion() {
++  return g_available_portal_version;
 +}
 +
- // static
- void SelectFileDialogLinuxPortal::DestroyPortalConnection() {
-   dbus_thread_linux::GetTaskRunner()->PostTask(
-@@ -214,6 +232,8 @@ void SelectFileDialogLinuxPortal::SelectFileImpl(
-                      weak_factory_.GetWeakPtr()));
-   info_->type = type;
-   info_->main_task_runner = base::SequencedTaskRunner::GetCurrentDefault();
-+  info_->button_label = button_label();
-+  info_->allow_multiple_selection = allow_multiple_selection();
```

Comment:
It looks like some handling around multiple selection was removed in this commit. https://github.com/electron/electron/commit/82e5c7421437c9e3e08eab20dcd4a28637ad8820.  Was that intentional?  Additionally, this patch has changed in main https://github.com/electron/electron/pull/45373 and this PR should be rebased to take those changes into account.

Reply:
I tested the latest build from this PR - multiselection works properly on Linux.